src/Nuget.PackageIndex/SettingsJson.cs

```diff
@@ -0,0 +1,182 @@
+    /// Represents settings.json file located in the index directory. If file does not
+    /// exist, creates a new settings.json with default settings. When user modifies, settings.json
+    /// changes are in effect after VS is re-opened.
+    /// </summary>
+    public class SettingsJson : ISettingsJson
+    {
+        private const string SettingJsonFileName = "settings.json";
+        private const string DefaultJsonContent = @"{
+    'includePackagePatterns' : []
+}
+";
+        /// <summary>
+        /// These are prefixes for package names that are shipped under,
+        /// %ProgramFiles(x86)%\Microsoft Web Tools\DNU. We do look for packages 
+        /// at multiple local folders (sources), but index only currated list of 
+        /// packages. If users want to add other packages to the index they could 
+        /// update file %UserProfile%\AppData\Local\Microsoft\PackageIndex\settings.json
+        /// and add a regex for their packages to "include" list.
+        /// </summary>
+        private readonly string[] _defaultIncludeRules = new []
+        {
+            @"dnx\..+",
+            @"entityframework\..+",
+            @"entityframework7\..+",
+            @"microsoft\..+",
+            @"newtonsoft\..+",
+            @"system\..+",
+            @"remotion\..+"
+        };
+
+        private readonly IFileSystem _fileSystem;
+        private string _settingsJsonPath;
+
+        public SettingsJson(string indexDirectory)
+            : this(indexDirectory, new FileSystem())
+        {
+
+        }
+
+        internal SettingsJson(string indexDirectory, IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+            _settingsJsonPath = Path.Combine(indexDirectory, SettingJsonFileName);
+        }
+
+        private JObject InitializeSettingsJson()
+        {
+            JObject jObject = null;
+            if (_fileSystem.FileExists(_settingsJsonPath))
+            {
+                jObject = ReadJsonFromFile(_settingsJsonPath);
+            } 
+            else
+            {
+                jObject = GenerateDefaultSettingsJson();
+                WriteJsonToFile(jObject);
+            }
+
+            return jObject;
+        }
+
+        private JObject GenerateDefaultSettingsJson()
+        {
+            var jObject = JObject.Parse(DefaultJsonContent);
+            JArray includePackagePaterns = (JArray)jObject["includePackagePatterns"];
+            foreach(var pattern in _defaultIncludeRules)
+            {
+                includePackagePaterns.Add(pattern);
+            }
+
+            return jObject;
+        }
+
+        private JObject ReadJsonFromFile(string filePath)
+        {
+            JObject result = null;
+
+            try
+            {
+                var fileContent = _fileSystem.FileReadAllText(filePath);
+                result = JObject.Parse(fileContent);
+            }
+            catch (Exception e)
+            {
+                Debug.Write(e.ToString());
```

Comment:
Just emit a warning. You should still catch


Ship it from me once comments are clean. You want another set of eyes on this 

-----Original Message-----
From: "Anton Piskunov" <notifications@github.com>
Sent: ‎7/‎16/‎2015 7:18 AM
To: "NuGet/NuGet.PackageIndex" <NuGet.PackageIndex@noreply.github.com>
Cc: "Yishai Galatzer" <yishai_galatzer@yahoo.com>
Subject: Re: [NuGet.PackageIndex] Support filtering sets of packages and move suggestion to the end of … (#13)

In src/Nuget.PackageIndex/SettingsJson.cs:
> +
> +            return jObject;
> +        }
> +
> +        private JObject ReadJsonFromFile(string filePath)
> +        {
> +            JObject result = null;
> +
> +            try
> +            {
> +                var fileContent = _fileSystem.FileReadAllText(filePath);
> +                result = JObject.Parse(fileContent);
> +            }
> +            catch (Exception e)
> +            {
> +                Debug.Write(e.ToString());
it would crash VS, i can not do that. Also that settings.json can be edited by two VS instances simultaneously - rare , but possible. This whole design might change soon and we are not advertising settings.json yet to users 
—
Reply to this email directly or view it on GitHub.